WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Notes Section/Customer Note/EditCustomerNoteViewModel.swift

```diff
@@ -40,42 +50,40 @@ final class EditCustomerNoteViewModel: EditCustomerNoteViewModelProtocol {
     ///
     private let analytics: Analytics
 
-    init(order: Order, stores: StoresManager = ServiceLocator.stores, analytics: Analytics = ServiceLocator.analytics) {
+    /// Service to check if a feature flag is enabled.
+    ///
+    private let featureFlagService: FeatureFlagService
+
+    init(order: Order,
+         stores: StoresManager = ServiceLocator.stores,
+         analytics: Analytics = ServiceLocator.analytics,
+         noticePresenter: NoticePresenter = ServiceLocator.noticePresenter,
+         featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
         self.order = order
         self.newNote = order.customerNote ?? ""
         self.stores = stores
         self.analytics = analytics
+        self.noticePresenter = noticePresenter
+        self.featureFlagService = featureFlagService
         bindNavigationTrailingItemPublisher()
     }
 
+    func update(order: Order) {
+        self.order = order
+    }
+
     /// Update the note remotely and invoke a completion block when finished
     ///
     func updateNote(onFinish: @escaping (Bool) -> Void) {
-        let modifiedOrder = order.copy(customerNote: newNote)
-        let action = OrderAction.updateOrder(siteID: order.siteID, order: modifiedOrder, fields: [.customerNote]) { [weak self] result in
-            guard let self = self else { return }
-
-            self.performingNetworkRequest.send(false)
-            switch result {
-            case .success:
-                self.presentNotice = .success
-                self.analytics.track(event: WooAnalyticsEvent.OrderDetailsEdit.orderDetailEditFlowCompleted(subject: .customerNote))
-            case .failure(let error):
-                self.presentNotice = .error
-                self.analytics.track(event: WooAnalyticsEvent.OrderDetailsEdit.orderDetailEditFlowFailed(subject: .customerNote))
-                DDLogError("⛔️ Unable to update the order: \(error)")
-            }
-
-            onFinish(result.isSuccess)
-        }
-
-        performingNetworkRequest.send(true)
-        stores.dispatch(action)
+        dispatchUpdateOrderOptimisticallyAction(withNote: newNote, onFinish: onFinish)
     }
 
     /// Track the flow cancel scenario.
     ///
     func userDidCancelFlow() {
+        // We need to set the original value.
+        syncNewNoteWithOrder()
```

Comment:
As we are using a unique view model, we have to reset the value when the flow is cancelled and the modal is dismissed

Reply:
yep, I agree with you about separating the logic 👍🏼 your suggestion sounds good to me, let me check it out
Thanks!